apps/web/src/components/ui/button-with-disable-tooltip.tsx

```diff
@@ -0,0 +1,35 @@
+import { Button } from "./button";
+import type { ButtonProps } from "./button";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "./tooltip";
+
+interface ButtonWithDisableTooltipProps extends ButtonProps {
+  tooltip: string;
+}
+
+export const ButtonWithDisableTooltip = ({
+  tooltip,
+  disabled,
+  ...props
+}: ButtonWithDisableTooltipProps) => {
+  const ButtonComponent = <Button {...props} disabled={disabled} />;
+
+  if (disabled) {
+    // If the button is disabled, we wrap it in a tooltip since
+    // we don't want to show the tooltip if the button is enabled.
+    return (
+      <TooltipProvider>
+        <Tooltip>
+          <TooltipTrigger>{ButtonComponent}</TooltipTrigger>
+          <TooltipContent>{tooltip}</TooltipContent>
+        </Tooltip>
+      </TooltipProvider>
+    );
+  }
```

Comment:
We are missing the `asChild`, the `span` and the `"pointer-events-none"` classname (maybe use `cn` to append it if disabled) for the disabled component (see [radix-ui](https://www.radix-ui.com/primitives/docs/components/tooltip#displaying-a-tooltip-from-a-disabled-button)). Once done, we can ship it!

Reply:
Worked on that, please check it up and let me know what you think